src/App.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from "react";
+import Button from "./components/Button";
+
+function App() {
+    const submit = e => {
+        e.preventDefault();
+        setUpdating(false);
+        const tip = (total.value / 100) * percentage.value;
+        setTip(tip);
+    };
+
+    function useControlledInput(initial) {
+        const [value, setValue] = useState(initial);
+        return {
+            value,
+            onChange: e => {
+                setUpdating(true);
+                setValue(parseInt(e.target.value || 0, 10));
+            }
```

Comment:
Using an arrow function within a handler function isn't super readable. You might want to write this function elsewhere and invoke it here instead.